app/views/layouts/_header.html.erb

```diff
@@ -0,0 +1,28 @@
+
+<% if current_user %>
+<nav class="navbar navbar-expand-lg bg-light">
+<div class="container-fluid">
+   <a class="navbar-brand" href="#">Navbar</a>
+   <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
+      <span class="navbar-toggler-icon"></span>
+   </button>
+   <div class="collapse navbar-collapse" id="navbarSupportedContent">
+   <ul class="navbar-nav ms-auto mb-2 mb-lg-0">
+      <li class="nav-item">
+      <%= link_to "Home" , root_path, class:"nav-link" %>
+      </li>
+      <li class="nav-item">
+      <%= link_to "Recipes" , recipes_path, class:"nav-link" %>
+      </li>
+      <li class="nav-item">
+      <%= link_to "Public Recipes" , public_recipes_path, class:"nav-link" %>
+      </li>
+      <li class="nav-item">
+      <%= link_to "General Shopping list" ,general_shopping_list_index_path, class:"nav-link" %>
+      </li>
+      <li class="nav-item">
+      <%= button_to "Log out", destroy_user_session_path, method: :delete, data: { turbo: false }, class: "logout_button", class:"nav-link"  %>
```

Comment:
- [ ] You have duplicated the key class here which is not a good practice when adding a class to an element. 
```diff
- <%= button_to "Log out", destroy_user_session_path, method: :delete, data: { turbo: false }, class: "logout_button", class:"nav-link" %>
+ <%= button_to "Log out", destroy_user_session_path, method: :delete, data: { turbo: false }, class: "logout_button nav-link",  %>
```